editor/import/scene_import_settings.cpp

```diff
@@ -534,6 +549,30 @@ void SceneImportSettings::update_view() {
 	update_view_timer->start();
 }
 
+void SceneImportSettings::update_state() {
+	if (scene_import_settings_data->category == ResourceImporterScene::INTERNAL_IMPORT_CATEGORY_MATERIAL) {
+		MaterialData &md = material_map[selected_id];
+		if (md.settings["use_external/enabled"] && md.settings["use_external/path"]) {
+			material_preview->set_material(ResourceLoader::load(md.settings["use_external/path"]));
+		} else {
+			material_preview->set_material(md.material);
+		}
```

Comment:
I wonder if this logic should be moved to ``_set`` and then the rest of this function could be shared between ``_fill_scene`` and ``_scene``. As it stands there is a lot of code duplication and this function is only ever called in one place.

Reply:
That was also my concern when making this pull request. I'm currently at work, but I'll make another change for this when I'm home.